colossalai/inference/kv_cache/kvcache_manager.py

```diff
@@ -61,6 +61,12 @@ def __init__(self, config: InferenceConfig, model_config: PretrainedConfig, verb
         self.tp_size = config.tp_size
         # Model settings
         self.dtype = config.dtype
+
+        if config.kv_cache_dtype is None:
+            self.kv_cache_dtype = config.dtype
+        else:
+            self.kv_cache_dtype = config.kv_cache_dtype
```

Comment:
Branching can be skipped, as the above comment indicated. `self.kv_cache_dtype = config.kv_cache_dtype`